content/400-reference/200-api-reference/100-prisma-schema-reference.mdx

```diff
@@ -1319,18 +1336,31 @@ The name of the `fields` argument on the `@@id` attribute can be omitted:
 @@id([title, author])
 ```
 
+**Note:** In version 3.5.0 and later, new `length` and `sort` arguments are available as a Preview feature. See [Index configuration](../../../../concepts/components/prisma-schema/index-configuration) for more details.
+
+| Name     | Required | Type     | Description                                                                                                                 |
+| -------- | -------- | -------- | --------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                          |
+| `sort`   | **No**   | `String` | **SQL Server only.** Allows you to specify in what order the entries of the index or constraint are stored in the database. |
+
 #### Signature
 
 ```prisma no-lines
 @@id(_ fields: FieldReference[], name: String?, map: String?)
 ```
 
-> **Note**: Until version 3.0.0, the signiture was:
+> **Note**: Until version 3.0.0, the signature was:
 >
 > ```prisma no-lines
 > @@id(_ fields: FieldReference[])
 > ```
 
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
+>
+> ```prisma no-lines
+> @id(_ fields: FieldReference[], name: String?, map: String?, length: number?, sort: String?)
```

Comment:
The signature is a bit different on compounds since length and sort are defined on the field level: 
`@@id(_ fields: FieldReference[], name: String?, map: String?)`
With field reverence now being 
`filename(length: Number?, sort: String?)` If no option is given, no empty braces are rendered, just the name.
Example: `@@id([firstName, middleName(length: 1), lastName(sort:Desc)], name: "fullName", map: "PkConstraint")`

Reply:
Thanks, would this mean that the signature of `@@id` is unchanged for the preview feature, but the signature of `FieldReference[]` has? So should I update that somewhere else?

Also I don't quite follow the 'If no option is given' section, where would the option be given?

Reply:
Yes, I think you're correct, the @@id signature stays the same. 
No option given refers to length or sort, so in my example firstName has no additional options specified and stays as `firstName` instead of `firstName()`.

Reply:
Thanks for clearing that up. I've left the signature as unchanged.